demo/admin/src/products/categories/AssignProductsGrid.tsx

```diff
@@ -0,0 +1,80 @@
+import { gql, useApolloClient, useQuery } from "@apollo/client";
+import { Savable } from "@comet/admin";
+import { CircularProgress } from "@mui/material";
+import {
+    GQLGetProductIdsForProductCategoryQuery,
+    GQLGetProductIdsForProductCategoryQueryVariables,
+    GQLSetProductCategoryMutation,
+    GQLSetProductCategoryMutationVariables,
+} from "@src/products/categories/AssignProductsGrid.generated";
+import { ProductsSelectGrid } from "@src/products/categories/ProductsSelectGrid";
+import isEqual from "lodash.isequal";
+import React, { useEffect, useState } from "react";
+import { FormattedMessage } from "react-intl";
+
+const setProductCategoryMutation = gql`
+    mutation SetProductCategory($id: ID!, $input: [ID!]!) {
+        updateProductCategory(id: $id, input: { products: $input }) {
+            id
+        }
+    }
+`;
+
+const getProductIdsForProductCategory = gql`
+    query GetProductIdsForProductCategory($id: ID!) {
+        products(filter: { category: { equal: $id } }) {
+            nodes {
+                id
+            }
+        }
+    }
+`;
+
+interface FormProps {
+    productCategoryId: string;
+}
+
+export function AssignProductsGrid({ productCategoryId }: FormProps): React.ReactElement {
+    const client = useApolloClient();
+
+    const { data, error, loading } = useQuery<GQLGetProductIdsForProductCategoryQuery, GQLGetProductIdsForProductCategoryQueryVariables>(
+        getProductIdsForProductCategory,
+        {
+            variables: { id: productCategoryId },
+        },
+    );
+    const initialValues = data?.products.nodes ? data.products.nodes.map((product) => product.id).sort() : [];
+    const [values, setValues] = useState<string[]>(initialValues);
+    useEffect(() => {
+        setValues(initialValues);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [data?.products.nodes]);
```

Comment:
Why do you need this effect? Disabling the ESLint rule doesn't seem like a good idea to me.

Reply:
because fetched values need to be in values-state to allow deselecting. `data.products.node` is not set for first render so the default-value for values-useState is an empty array and is not updated after `data.products.node` is finally fetched. es-lint requires useEffect to contain initialValues in it's deps-array, but this is always newly initialized resulting in endless re-render.

Reply:
It would be better to do this this way:

```ts
const initialValues = useMemo(() => {
  return data?.products.nodes ? data.products.nodes.map((product) => product.id).sort() : [];
}, [data]);

useEffect(() => {
  setValues(initialValues)
}, [initialValues]);
```

Or, remove the useEffect altogether:

```ts
const initialValues = data?.products.nodes ? data.products.nodes.map((product) => product.id).sort() : [];

if (!isEqual(initialValues, values)) {
  setValues(initialValues);
}
```

BTW, this is a conservation I would like to resolve 😉

Reply:
tried your second idea but it didn't work because !isEqual resulted in always reseting back to initialValues, so used your first suggestion.